client/app/components/contact-form.js

```diff
@@ -0,0 +1,43 @@
+import Ember from 'ember';
+import config from '../config/environment';
+const isTest = config.APP.STUB_EXTERNAL_MOCK;
+
+export default Ember.Component.extend({
+
+  districtContactEmail: '',
+  districtContactSubmitted: false,
+  districtContactErrored: false,
+
+  getSubmittedName: Ember.computed('name', function() {
+    return this.get('name') != undefined ? `${this.get('name')} ,` : '';
+  }),
+
+  actions: {
+    submitDistrictContact() {
+      this.set('districtContactErrored', false);
+
+      if (Ember.isPresent(this.get('districtContactEmail'))) {
+
+        Ember.$.ajax({
+          url: 'https://formspree.io/schoolbot-contact@vermonster.com',
+          method: 'POST',
+          dataType: 'json',
+          data: {
+            _replyto: this.get('districtContactEmail'),
+            _subject: `District contact by ${this.get('districtContactEmail')}`,
+            name: this.get('name'),
+            message: this.get('message')
+          }
+        }).then(() => {
+          this.set('districtContactSubmitted', true);
+        }).catch(() => {
+          if (isTest) {
```

Comment:
We may not even need to put test code in Ember at all – there is [puffing-billy](https://github.com/oesmith/puffing-billy) for mocking ajax responses from RSpec, which would also allow us to easily test both success and failure scenarios. Given the customizations to our test environment it may not be a drop-in thing though.